### DISCUSSION TITLE
3.1.4. Updating Meetings through TSML UI

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#772] [Q&A] [answered]
I was in the habit of updating meetings by bringing up the meeting in the front end of WordPress and then select edit meeting. Using the new UI isn't as easy.

Though I know many users seem to like to do mass updates by uploading CSV files, I have been doing them directly through WordPress. For some reason that I cannot recall, I was told this was the better way of updating the meetings.

Using the WordPress dashboard is clunky to find the meeting to edit. There's just a limited amount of fields to sort by: meeting name, day, time, region, date published (which gets changed every time an import is done). It was so much easier to go to the meeting page, select the day I was looking for and then select the meeting to edit.

Is there any plan in the works to make editing meetings easier through WordPress?

**Comment by joshreisner** [comment, marked as answer]
sounds like a good feature. it should be possible with some TSML javascript to watch the query string and add/update a link in the admin toolbar.

before we do that, if you're comfortable opening the javascript console, i've [created a ticket](https://github.com/code4recovery/tsml-ui/issues/225) to log the edit_url when present

before that works in TSML, we're also going to need [this ticket](https://github.com/code4recovery/12-step-meeting-list/issues/438) to be completed

**Reply by Grateful-Viv** [reply]
I will have some time later this afternoon (it's 1130 here on the right coast). Just tell me what you need me to do. I've never opened a JavaScript console, but am willing to give it a try. On Windows in Chrome it seems to be  Control + Shift + J

https://developer.chrome.com/docs/devtools/open/#:~:text=%23%20Open%20the%20Console%20panel%20to,Get%20Started%20With%20The%20Console.

**Reply by Grateful-Viv** [reply]
@joshreisner I figured out how to open a JavaScript console. Any time now is good. Thanks.

**Reply by joshreisner** [reply]
thats correct! i still need to do some programming but in another ticket im proposing i log a url there that you could click to jump to the edit screen

**Reply by joshreisner** [reply]
@Grateful-Viv as of version `3.14.3` if you open the javascript console on a TSML UI meeting detail screen you should see a link to the meeting edit page. 

<img width="1476" alt="Screen Shot 2022-05-14 at 6 01 57 PM" src="https://user-images.githubusercontent.com/1551689/168453069-129e07bc-bd76-405d-b738-5befcddfa2ce.png">

**Reply by Grateful-Viv** [reply]
@joshreisner, that works. I'd rather do this for a couple of changes to meetings than either switch back to Legacy or have to download CSV, make the change(s) and the upload again. Every time that is done, all draft meetings are wiped out. We keep a few seasonal meetings in draft status. Though putting them in the trash and not emptying it, does keep the meetings even after an upload.

Are there plans to make it have a nicer way instead of opening the JavaScript console? While I don't really mind doing this, I think others may find it cumbersome. 

Thanks.

**Reply by joshreisner** [reply]
ok, i've added edit links now. hopefully this will be a nicer experience for wordpress admins

![Screen Shot 2022-05-16 at 7 48 38 AM](https://user-images.githubusercontent.com/1551689/168622566-edaee88d-6b56-4382-9fcc-6e7cc6af8af4.png)

**Reply by Grateful-Viv** [reply]
That’s awesome and so much appreciated. Thanks, @joshreisner!

**Reply by Grateful-Viv** [reply]
Just noticed your screen shot. Switching the plugin to TSML UI and going to /meetings/ does not have the same options as if I were on a page using the react shortcode. [tsml_react]

Below is with TSML UI enabled through the plugin. Notice there's no edit page option.
![Meetings-NoShortcode](https://user-images.githubusercontent.com/62247421/168660966-04a70956-01a3-4c18-a954-75bab01ee80f.png)

Below is going through a page that has the shortcode. That was the way I've been testing TSML UI for over a year. It has an option to edit the page. Problem is the widgets aren't working with the latest plugin update on a "page" , but rather through the plugin Settings.

![Meetings-Shortcode](https://user-images.githubusercontent.com/62247421/168662522-2dc4bb14-5b3d-4531-a7a3-16499851a993.png)

**Reply by joshreisner** [reply]
that's correct. when you use it as a shortcode you are on a page, when you are using the TSML display setting you are on what wordpress calls a "custom post type archive" - this is why you can't edit it like you would a page

our custom post type archive has support for our TSML widget areas, pages by default do not.

**Reply by Grateful-Viv** [reply]
Does that mean we won’t see an edit button? Maybe I’m the only one wanting to edit meetings this way. I can do the JavaScript console, just not sure others are going to want to.

**Reply by jcookson-infoblox** [reply]
I have had a few occurrences in the last few weeks where this would have been nice to have in place.

**Reply by joshreisner** [reply]
i was answering your question about the `Edit Page` button. in either case you should see an `Edit Meeting` button when looking at a meeting detail page.

**Comment by brianw-area24** [comment]
We also have an issue I opened a while back to add additional filter(s) in the Wordpress dashboard page. See #370.  If you'd like to suggest field(s) to filter by, please comment on that issue.

**Comment by jcookson-infoblox** [comment]
is this about adding ...
<img width="284" alt="Screen Shot 2022-05-14 at 10 35 13 AM" src="https://user-images.githubusercontent.com/52213936/168449457-b2135d7a-526c-41bd-b5cb-a64a1294ca07.png">
in the meeting UI for meeting admins?

**Reply by joshreisner** [reply]
thanks for prompting me to follow up, see [my reply here](https://github.com/code4recovery/12-step-meeting-list/discussions/772#discussioncomment-2752535)
